corehq/apps/auditcare/middleware.py

```diff
@@ -17,75 +17,37 @@ def __init__(self, get_response=None):
         is instantiated on the request object.
         """
         super(AuditMiddleware, self).__init__(get_response)
-        self.active = False
-        self.log_admin = True
-
-        if hasattr(settings, "AUDIT_ADMIN_VIEWS"):
-            self.log_admin = settings.AUDIT_ADMIN_VIEWS
+        self.active = any(getattr(settings, name, False) for name in [
+            "AUDIT_ALL_VIEWS", "AUDIT_VIEWS", "AUDIT_MODULES"
+        ])
+        self.audit_modules = tuple(settings.AUDIT_MODULES)
+        self.audit_views = set(settings.AUDIT_VIEWS)
```

Comment:
Not sure I follow the _settings.py_ thing -- do you just mean that both of those settings currently have assigned values? If so, I think this implementation breaks encapsulation -- I'm required to have knowledge of what _settings.py_ is doing, and someone might remove those values in the future, not realizing that this module depends on those settings existing.

I think the thing that nags me about the tuple/set thing is that `set` is unnecessary -- `in` is available for tuples, lists, and sets. So when it is explicitly defined here, it tells me that unique elements are important. And then that raises the question of why unique elements aren't important for `audit_modules`. I'm probably overthinking this, though.

Reply:
> someone might remove those values in the future, not realizing that this module depends on those settings existing.

That would break the test code
https://github.com/dimagi/commcare-hq/blob/56df3ee2ac088f40dd43df6e6fd8641aedd320b9/corehq/apps/auditcare/tests/test_middleware.py#L133-L134

> `set` is unnecessary -- `in` is available for tuples, lists, and sets.

Member lookup for sets is O(1), for lists and tuples it's O(N)